out_writeapi_test.go

```diff
@@ -38,3 +44,143 @@ func TestFLBPluginRegister(t *testing.T) {
 	assert.Equal(t, "writeapi", currplugin.name)
 
 }
+
+type StreamChecks struct {
+	calledGetContext     bool
+	calledcheckResponses bool
+	createDecoder        bool
+	gotRecord            bool
+	calledparseMap       bool
+	calledJTB            bool
+	appendRows           bool
+	appendQueue          bool
+}
+
+type MockManagedStream struct {
+	managedstream  *managedwriter.ManagedStream
+	AppendRowsFunc func(ctx context.Context, data [][]byte, opts ...managedwriter.AppendOption) (*managedwriter.AppendResult, error)
+	StreamName     func() string
+	CloseFunc      func() error
+}
+
+func (m *MockManagedStream) AppendRows(ctx context.Context, data [][]byte, opts ...managedwriter.AppendOption) (*managedwriter.AppendResult, error) {
+	return m.AppendRowsFunc(ctx, data, opts...)
+}
+
+func (m *MockManagedStream) Close() error {
+	return m.managedstream.Close()
+}
+
+func TestFLBPluginFlushCtx(t *testing.T) {
+	var checks StreamChecks
+	var setID interface{}
+	var id int
+
+	mockMS := &MockManagedStream{
+		AppendRowsFunc: func(ctx context.Context, data [][]byte, opts ...managedwriter.AppendOption) (*managedwriter.AppendResult, error) {
+			checks.appendRows = true
```

Comment:
Can we perform verification of the expected payload here?

Reply:
You mean check the payload of the data?

Reply:
Yes it should be in protobuf format and contain the expected fields.